ripc-protocol-tcp/src/main/java/io/ripc/protocol/tcp/TcpHandler.java

```diff
@@ -0,0 +1,8 @@
+package io.ripc.protocol.tcp;
+
+import org.reactivestreams.Publisher;
+
+public interface TcpHandler<R, W> {
+
+    Publisher<Void> handle(Connection<R, W> connection);
```

Comment:
This is a server construct. One server can only ever have a single handler.

> Does each Subscriber receive an onComplete signal when the peer terminates?

The only subscriber to this `Publisher<Void>` is reactive-ipc internals. No connection lifecycle events are piped to this, unless the handle implementation choose to do so. eg:

This does not wire the connection close.

```java
        Netty4TcpServer.create(0)
                       .start(connection -> Publishers.error(new IllegalStateException()));
```

OTOH, this does as it is wiring the write to the returned `Publisher` which fails when connection is closed:

```java
        Netty4TcpServer.create(0)
                       .start(connection -> connection.write(connection));
```